lib/matplotlib/markers.py

```diff
@@ -347,7 +372,10 @@ def get_transform(self):
         Return the transform to be applied to the `.Path` from
         `MarkerStyle.get_path()`.
         """
-        return self._transform.frozen()
+        if self._user_transform is not None:
```

Comment:
Instead of these `if` blocks, would it be possible to just set `_user_transform` to an identity transform if it's `None` in `__init__`?

Reply:
Yes, it is possible, or instead of defaulting to None, we can default to identity. However, I was thinking that check `is not None` will have less impact on performance than explicit __add__ with identity. Maybe it is a premature optimization.

Reply:
I did a quick check and I think this is an optimization worth making. It would reduce code duplicaiton to add a staticmethod that looks like this:
```python
def _combine_transforms(t1, t2):
    if t2 is None:
        return t1.frozen()
    else:
        return (t1 + t2).frozen()
```
which can then be used anywhere you want to combine two transformations where the second one can be None.

Reply:
I have added a private function, but I have found only two use cases, maybe I am missing some :worried: